pandas/io/parsers.py

```diff
@@ -2515,6 +2515,11 @@ def _rows_to_cols(self, content):
 
             msg = ('Expected %d fields in line %d, saw %d' %
                    (col_len, row_num + 1, zip_len))
+            if len(self.delimiter) > 1 and self.quoting != csv.QUOTE_NONE:
```

Comment:
Is the second part of the check needed (`self.quoting != csv.QUOTE_NONE`). Because AFAIU also when you do pass this, quotes are still be ignored by the regex expression to split the line, and you can still have this problem.